### ISSUE TITLE
How to use independent test sets

### ISSUE THREAD

**Issue by YinZhe0412** [mahmoodlab/CLAM#166]
Hello,
When using independent test sets, I don't know how to add the contents of my own datasets. I have tried to add features directory into --data_root_dir, modified the csv file in eval.py, but there is something wrong.

CUDA_VISIBLE_DEVICES=0 python eval.py --drop_out --k 10 --models_exp_code task_1_tumor_vs_normal_CLAM_50_s1 --save_exp_code task_1_tumor_vs_normal_CLAM_50_s1_cv --task task_1_tumor_vs_normal --model_type clam_sb --results_dir results --data_root_dir lung_FEATURES_DIRECTORY --split all

lung_FEATURES_DIRECTORY：features of my test set
task_1_tumor_vs_normal_CLAM_50_s1: results of trained model

Thank you!

**Comment by fedshyvana** [comment]
sorry for the late reply, if you still have not figured out, maybe you could post your modification to eval.py and the error you're getting and i can try to take a look. Feel free to re-open the issue in that case

**Comment by YinZhe0412** [comment]
The eval.py has not been modified，but the file csv is changed to Step_3.csv.
[Step_3.csv](https://github.com/mahmoodlab/CLAM/files/10683155/Step_3.csv)
The error is that the later independent test set is used for testing, but the pt file of the training set appears (can be found in Step_3. csv). This is something I can't understand. Sorry, I haven't been in this field for a long time.

The error:
![image](https://user-images.githubusercontent.com/81342165/217471593-cb82006b-83d2-4ccd-be5d-7eed3f0bbe70.png)

task_1_tumor_vs_normal_CLAM_50_s1:
![image](https://user-images.githubusercontent.com/81342165/217476017-ee01c447-fbd5-4686-8731-c897d5daaff9.png)

splits_0.csv:
![image](https://user-images.githubusercontent.com/81342165/217475740-642b7703-810f-4802-be4f-5e9a7421c096.png)

Thank you!
